File: Backend/Schedule/Data/Models/DTO/LessonDto.cs
namespace Schedule.Data.Models.DTO
{
    public class LessonDto
    {
        public Guid Id { get; set; }
        public NumberLesson NumberLesson { get; set; }
        public DayOfTheWeek Day { get; set; }
        public TypeLesson Type { get; set; }
        public DateTime StartPeriodDate { get; set; }
        public DateTime EndPeriodDate { get; set; }
        public SubjectDto Subject { get; set; }
        public GroupDto Group { get; set; }
        public TeacherDto Teacher { get; set; }
        public AudienceDto Audience { get; set; }
    }
}

File: app/src/main/java/com/example/schedule/model/dto/LessonDto.java
package com.example.schedule.model.dto;

import com.example.schedule.model.entity.DayOfTheWeek;
import com.example.schedule.model.entity.NumberLesson;
import com.example.schedule.model.entity.TypeLesson;

import org.joda.time.DateTime;

public class LessonDto {
    public String id;
    public SubjectDto subject;
    public GroupDto group;
    public TeacherDto teacher;
    public AudienceDto audience;
    public int numberLesson;

    public String startPeriodTime;
    public String endPeriodTime;

    public LessonDto(String subjectName, String groupNumber, String teacherName, String audienceNumber) {
        subject = new SubjectDto(subjectName);
        group = new GroupDto(groupNumber);
        teacher = new TeacherDto(teacherName);
        audience = new AudienceDto(audienceNumber);
    }

    public LessonDto(){};
}
